internal/interfaces/protocols_s3_policy.go

```diff
@@ -0,0 +1,269 @@
+package interfaces
+
+import (
+    "fmt"
+    "strings"
+
+    "github.com/hashicorp/terraform-plugin-framework/types"
+    "github.com/mitchellh/mapstructure"
+    "github.com/netapp/terraform-provider-netapp-ontap/internal/restclient"
+    "github.com/netapp/terraform-provider-netapp-ontap/internal/utils"
+)
+
+// ProtocolsS3PolicyGetDataModelONTAP describes the GET data model using go types for mapping.
+type ProtocolsS3PolicyGetDataModelONTAP struct {
```

Comment:
We have some existed data type you can reuse. You may take a look how this kind data type defined. For example, [this one](https://github.com/NetApp/terraform-provider-netapp-ontap/blob/integration/main/internal/interfaces/storage_flexcache.go#L13)